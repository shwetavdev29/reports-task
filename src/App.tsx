import React from "react";
import "bootstrap/dist/css/bootstrap.min.css";
import "@trendmicro/react-sidenav/dist/react-sidenav.css";
import "./App.css";
import { Reports } from "./pages";
import { MainLayout } from "./layouts";

function App() {
  return (
    <div className="App">
      <MainLayout>
        <Reports />
      </MainLayout>
    </div>
  );
}

export default App;
