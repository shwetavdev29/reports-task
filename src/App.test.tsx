import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  const {asFragment} = render(<App />);
  const linkElement = screen.getByText(/Reports/i);
  expect(linkElement).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
