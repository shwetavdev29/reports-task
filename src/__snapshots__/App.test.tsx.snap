// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <nav
        class="shadow-sm bg-white navbarStyles navbar navbar-expand navbar-light bg-light"
      >
        <div
          class="container-fluid"
        >
          <span
            class="navbar-brand"
          >
            <img
              alt="logo"
              class="logoImg"
              src="logo.svg"
            />
          </span>
          <div
            class="flex w-100 justify-content-start navbar-nav"
          >
            <div
              class="nav-item"
            >
              <img
                alt="menu"
                class="logoImg"
                role="button"
                src="hamburger.svg"
              />
            </div>
          </div>
          <div
            class="d-flex w-25 flex-row justify-content-end align-items-center"
          >
            <div
              class="profileIcon"
            >
              JD
            </div>
            <span
              class="ms-2 userName"
            >
              John Doe
            </span>
          </div>
        </div>
      </nav>
    </div>
    <div
      class="d-flex"
    >
      <nav
        class="h-100  bg-white reportsidenav sidenav---sidenav---_2tBP sidenav---collapsed---LQDEv"
        style="display: none; position: relative;"
      >
        <div
          class="mt-4 pt-3 bg-white sidenav---sidenav-nav---3tvij"
          role="menu"
        >
          <div
            class="sidenav---sidenav-navitem---uwIJ- sidenav---selected---1EK3y sidenav---highlighted---oUx9u"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="MENU_ICON1"
                class="not_allowed"
                src="menuicon1.svg"
              />
            </div>
          </div>
          <div
            class="sidenav---sidenav-navitem---uwIJ- sidenav---selected---1EK3y sidenav---highlighted---oUx9u"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="MENU_ICON2"
                class="not_allowed"
                src="menuicon2.svg"
              />
            </div>
          </div>
          <div
            class="sidenav---sidenav-navitem---uwIJ- sidenav---selected---1EK3y sidenav---highlighted---oUx9u"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="MENU_ICON3"
                class="not_allowed"
                src="menuicon3.svg"
              />
            </div>
          </div>
          <div
            class="sidenav---sidenav-navitem---uwIJ- sidenav---selected---1EK3y sidenav---highlighted---oUx9u"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="MENU_ICON4"
                class="pointer"
                src="menuicon4.svg"
              />
            </div>
          </div>
          <div
            class="sidenav---sidenav-navitem---uwIJ- sidenav---selected---1EK3y sidenav---highlighted---oUx9u"
            role="presentation"
          >
            <div
              class="sidenav---navitem---9uL5T"
              role="menuitem"
              tabindex="-1"
            >
              <img
                alt="MENU_ICON5"
                class="not_allowed"
                src="menuicon5.svg"
              />
            </div>
          </div>
        </div>
      </nav>
      <main
        class="pagewrapper"
      >
        <div
          class="mt-5 container-fluid"
        >
          <div
            class="row"
          >
            <div
              class="col-4"
            >
              <h3
                class="pageTitle"
              >
                 Reports
              </h3>
              <p
                class="reportDetail"
              >
                Easily generate a report of your transactions
              </p>
            </div>
            <div
              class="dropmain col-8"
            >
              <select
                class="reportDetailDropdown form-select"
                data-testid="select-project"
              >
                <option
                  class="dropitem"
                  value=""
                >
                  All Projects
                </option>
              </select>
              <select
                class="reportDetailDropdown form-select"
              >
                <option
                  class="dropitem"
                  value=""
                >
                  All Gateways
                </option>
              </select>
              <div
                class="reportDateSelect input-group"
              >
                <input
                  class="form-control"
                  type="date"
                />
              </div>
              <div
                class="reportDateSelect input-group"
              >
                <input
                  class="form-control"
                  type="date"
                />
              </div>
              <button
                class="generateReportBtn"
                data-testid="generate-report-btn"
              >
                Generate report
              </button>
            </div>
          </div>
          <div
            class="pe-3 row"
          >
            <div
              class="col-md-12 col-sm-12"
            >
              <div
                class="reportCard card"
              >
                <h3
                  class="cardTitle"
                >
                  All Projects | All Gateways
                </h3>
                <div
                  class="projectAccordion accordion"
                />
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-12"
            >
              <div
                class="amountCard card"
              >
                <h3
                  class="cardTitle"
                >
                  TOTAL: 0 USD
                </h3>
              </div>
            </div>
            <div
              class="col-12"
            >
              <p
                class="termstext"
              >
                Terms&Conditions | Privacy policy
              </p>
            </div>
          </div>
        </div>
      </main>
    </div>
  </div>
</DocumentFragment>
`;
